### ISSUE TITLE
Warnings are not treated as errors even if all settings are set correctly

### ISSUE THREAD

**Issue by hasrthur** [dotnet/roslyn#57539]
Hi! Hope you are doing great!

I faced a problem with my project where I wanted all analyzers warnings to be errors. I simply can't make all warnings into errors.

To reproduce an issue I created a simple project: https://github.com/hasrthur/DotNetCoreWarningsTest.

**Version Used**:  6.0.100-rc.2.21505.57

**Steps to Reproduce**:

1.
`.editorconfig` has a settings to treat all analyzers violations as errors.

```
dotnet_analyzer_diagnostic.severity = error
```

2.
`*.csproj` file has the same setting set

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
    <AnalysisLevel>latest</AnalysisLevel>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <NoWarn>CA1014,CA1812</NoWarn>
  </PropertyGroup>

</Project>

```
3.

`Program.cs` contains redundant `using` statement

```c#
using System; // <- this should trigger error
// See https://aka.ms/new-console-template for more information
Console.WriteLine(42);
```

Rider shows that line as error.

But when I build a project either via rider or console the build succeeds.

Even if I add

```
dotnet_diagnostic.IDE0005.severity = error
```

to `.editorconfig` it has no effect at build time but when I add

```
dotnet_diagnostic.IDE0005.severity = none
```

the warning/error disappears in rider

**Expected Behavior**:
Build fails with error

**Actual Behavior**:
Build is successful without any warning

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/41640#issuecomment-985780130
